rudder-core/src/main/scala/com/normation/rudder/domain/reports/ExpectedReports.scala

```diff
@@ -145,9 +206,327 @@ case class ComponentExpectedReport(
+        ~ ( "nodeHeartbeatPeriod"    -> n.modes.nodeHeartbeatPeriod )
+        ~ ( "globalAgentRunInterval" ->
+                   (
+                       ( "interval"    -> n.modes.globalAgentRun.interval )
+                     ~ ( "startMinute" -> n.modes.globalAgentRun.startMinute )
+                     ~ ( "splayHour"   -> n.modes.globalAgentRun.startHour )
+                     ~ ( "splaytime"   -> n.modes.globalAgentRun.splaytime )
+                   )
+          )
+        ~ ( "nodeAgentRunInterval"   -> (n.modes.nodeAgentRun.flatMap { run =>
+            if(run.overrides.getOrElse(false)) {
+                   Some(
+                       ( "interval"    -> run.interval )
+                     ~ ( "startMinute" -> run.startMinute )
+                     ~ ( "splayHour"   -> run.startHour )
+                     ~ ( "splaytime"   -> run.splaytime )
+                   )
+            } else {
+              None
+            }
+          } ) )
+        )
+     ~  ("rules" -> (n.ruleExpectedReports.map { r =>
+           (
+             ("ruleId"     -> r.ruleId.value)
+           ~ ("serial"     -> r.serial)
+           ~ ("directives" -> (r.directives.map { d =>
+               (
+                 ("directiveId" -> d.directiveId.value)
+               ~ ("policyMode"  -> d.policyMode.map( _.name))
+               ~ ("isSystem"    -> d.isSystem )
+               ~ ("components"  -> (d.components.map { c =>
+                   (
+                     ("componentName" -> c.componentName)
+                   ~ ("cardinality"   -> c.cardinality)
+                   ~ ("values"        -> c.componentsValues)
+                   ~ ("unexpanded"    -> c.unexpandedComponentsValues)
+                   )
+                 }))
+               )
+             }))
+           )
+        }))
+    )
+  }
+
+  def parseJsonNodeExpectedReports(s: String): Box[JsonNodeExpectedReports] = {
+    import PolicyMode.{Audit, Enforce}
+    import net.liftweb.util.Helpers.tryo
+
+    implicit val formats = DefaultFormats
+
+    implicit class ToValidInt(b: BigInt) {
+      def toValidInt = if(b.isValidInt) b.toInt else throw new NumberFormatException(s"${b.toString} is not a valid integer")
+    }
+
+    def modes(json: JValue): Box[NodeModeConfig] = {
+      def fail = Failure(s"Cannot parse JSON as modes parameters for expected node configuration: ${compactRender(json)}")
+      (
+          complianceMode( json \ "globalComplianceMode")
+        , json \ "globalHeartbeatPeriod"
+        , agentRun( json \ "globalAgentRunInterval", None)
+        , policyMode( json \ "globalPolicyMode" \ "mode")
+        , json \ "globalPolicyMode" \ "overridable"
+        , policyMode(json \ "nodePolicyMode")
+      ) match {
+        case (Some(gcm), JInt(ghp), Some(gari), Some(gpm), JBool(gpo), npm) =>
+          try {
+            Full(NodeModeConfig(
+                GlobalComplianceMode(gcm, ghp.toValidInt)
+              , heartbeat( json \ "nodeHeartbeatPeriod")
```

Comment:
Maybe it could be out in the match above ... even if we match anything ... so we can have  an nhp varaible !!